app/models/shift.rb

```diff
@@ -1,6 +1,6 @@
 class Shift < ApplicationRecord
   has_paper_trail
-  belongs_to :user
+  belongs_to :user, foreign_key: :id, class_name: 'User'
```

Comment:
The foreign_key is _never_ going to be `:id`, unless you're doing something super wonky and uncool. You also do not have to specify the class name here because it is the default.
Let's break this down:
`belongs_to :user` means there is a column in the `shifts` table that is the `id` of a user.
`foreign_key: :id` specifies what this column is, and here you're telling rails that it's the `id` field. You're saying that the `id` field, then, is the `id` of a user. That is incorrect--every table in a database needs a unique identifying index, and that is, by default, the ID field--and there is no reason to change that in this application.
`class_name: 'User'` does no harm in this case but by saying `belongs_to :user` rails automatically assumes that the class name is `User`. You only have to specify the class name if the foreign key is something other than `user_id`.
tl;dr: put this back the way it was and add a `user_id` to the shifts table.